CRM/Civicase/Service/CaseTypeCategoryBaseEventHandler.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+use CRM_Civicase_Service_CaseTypeCategoryProcessor as CaseTypeCategoryProcessor;
+
+/**
+ * Class CRM_Civicase_Service_CaseTypeCategoryBaseEventHandler.
+ */
+class CRM_Civicase_Service_CaseTypeCategoryBaseEventHandler {
+
+  /**
+   * Run some OptionValue actions on some event.
+   *
+   * @param string $op
+   *   The operation being performed.
+   * @param string $objectName
+   *   Object name.
+   * @param mixed $objectId
+   *   Object ID.
+   * @param object|array $objectRef
+   *   Object reference.
+   * @param bool $loadFullData
+   *   (Optional) If TRUE then full case type category data will be loaded
+   *   from db. Is useful if $objectRef contains some partial data.
+   */
+  public function run($op, $objectName, $objectId, &$objectRef, $loadFullData = FALSE) {
```

Comment:
Why is a service class accepting parameters meant for a Hook ?
It is a service and should be usable by hooks and by any other class/function in civicase that needs to trigger a post create event for a case type category.